R/cyprus_readutils.R

```diff
@@ -0,0 +1,253 @@
+#' @description The CalcLoops code produces HDF5 files which contain
+#'              a matrix of momenta and the data for the loops (without
+#'              spin projection) organised by stochastic sample. Currently, the
+#'              reading code assumes that there is a single configuration stored per
+#'              file and the "trajectory" parameter in CalcLoops is assumed
+#'              to take its default value of '4'.
+#' @param selections Named list with names from the list 'Scalar', 'dOp', 'Loops'
+#'                   'LpsDw', 'LpsDwCv', 'LoopsCv' specifying the requesetd loop types. 
+#'                   The elements of this list are in turn expected
+#'                   be data frames of the form
+#'                     \tabular{rrr}{
+#'                       \strong{qx} \tab \strong{qy} \tab \strong{qz} \cr
+#'                       0           \tab 0           \tab 1           \cr
+#'                       -2          \tab 1           \tab -3          \cr
+#'                       ...         \tab ...         \tab ...
+#'                     }
+#'                   specifying the momentum combinations to be extracted for each
+#'                   loop type.
+#' @param files Vector of strings, list of HDF5 files to be processed.
+#' @param Time Integer, time extent of the lattice.
+#' @param nstoch Integer, number of stochastic samples to be expected in file.
+#' @param accumulated Boolean, specifies whether the loops, as organised by stochastic sample,
+#'                    are accumulated, such that, say, element \code{n} corresponds to the 
+#'                    sum over the first \code{n} stochastic samples. If specified as \code{TRUE},
+#'                    the data is post-processed to recover the measurements for the particular
+#'                    samples.
+#' @param legecy_traj Boolean. The root group for the loop data is 'conf_xxxx', where 'xxxx'
+#'                    corresponds to what is passed via the 'traj' flag to CalcLoops. When
+#'                    left empty, this defaults to '0004'. If this was left emtpy when
+#'                    the loop files were generated, set this to \code{TRUE} and the paths
+#'                    will be constructed with 'conf_0004' as their root group.
+#' @param verbose Boolean, output I/O time per file. Requires 'tictoc' package.
+#' @return Named list of the same length as \code{selections} containg the loop data
+#'         in the \link{raw_cf} format.
+#' @export
+cyprus_read_loops <- function(selections, files, Time, nstoch, accumulated = TRUE, legacy_traj = TRUE, verbose = FALSE){
+  rhdf5_avail <- requireNamespace("rhdf5")
+  dplyr_avail <- requireNamespace("dplyr")
+  if( !rhdf5_avail | !dplyr_avail ){
+    stop("The 'dplyr' and 'rhdf5' packages are required to use this function!\n")
+  }
+  if( verbose ){
+    tictoc_avail <- requireNamespace("tictoc")
+    if( !tictoc_avail ){
+      stop("Time reporting requires the 'tictoc' package!")
```

Comment:
so, this is a hard dependence on this package, isn't it?

Reply:
I think that this is similar to the HDF5 reading that we have. Most of the package can be used without these packages installed, so I think that this usage does not imply an `Imports` dependency but a mere `Suggests`.

Reply:
No, only if one wishes to use verbose mode.